index.md

```diff
@@ -88,13 +88,61 @@ Within this paradigm, we make the following decisions.
   - A linter can help you achieve that.
 
 
-### 2.2. Defensive programming
+### 2.2. Readability
+- Avoid variable hiding (i.e. declaring a variable with the same name as one which 
+already exist in scope)
+- Prefer `const` over `let`, and avoid reusing variable names unless it follows natural
+from the syntax (e.g. `for (let i; i < 10; i++) { ... }`)
+- Prefer `let` over `var`: Make your intention of how the variable should be used explicit
+- Make variable names as explicit as possible, and avoid abbreviations
+  - Exceptions are variables used as counter indexes (e.g. `[].map((item, i) => ...)` is ok
+  instead of `[].map((item, itemIndex) => ...)`)
+- Avoid use of deep scopes, e.g. instead of:
+  ```javascript
+  function foo (paramA) {
+    return function (paramB) {
+      function bar () {
+        // do stuff with paramA and paramB
+      }
+      
+      // call on function bar somewhere
+    }
+  }
+  ```
+  Prefer something closer to:
+  ```javascript
+  function foo (paramA) {
+    return function (paramB) {
+      // call on function bar somewhere with paramA and paramB
+    }
+  }
+  
+  function bar (paramA, paramB) {
+    // do stuff with paramA and paramB
+  }
+  ```
+  - If you have to use the former style, group the functions at the end 
```

Comment:
What would be possible/example reasons?

Reply:
It might just be a personal preference, but I find that injecting function declaration in the beginning or the middle of code interrupts my flow when reading code.

**Preferred code structure**

```javascript
// some code
function foo () {}
```

**Not preferred**

```javascript
function foo () {}
// some code
```

**Even worse (in my opinion)**

```javascript
// some code
function foo () {}
// more code
```